### PR TITLE
Refactor errors

### PR DESCRIPTION
**Pull request by severo**
Capture d’écran 2023-05-11 à 14 59 20

^ yes!

### update

Capture d’écran 2023-05-15 à 16 43 54

### CODE REVIEW

**Comment by severo** [comment]
yes, I will have some work...

File: services/api/src/api/utils.py [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
 ApiErrorCode = Literal[
-    "MissingRequiredParameter",
-    "InvalidParameter",
-    "ResponseNotReady",
-    "ResponseNotFound",
-    "UnexpectedError",
-    "ExternalUnauthenticatedError",
-    "ExternalAuthenticatedError",
     "AuthCheckHubRequestError",
+    "ExternalAuthenticatedError",
```
Comment (AndreaFrancis) [inline services/api/src/api/utils.py:15]: nit: Maybe take advantage of this PR and also move all API exceptions to a new file exceptions.py as in libcommon?

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 2]
```diff
@@ -218,11 +158,9 @@ def raise_if_blocked(
             The list of blocked datasets. If empty, no dataset is blocked.
     Returns:
         `None`
-    <Tip>
     Raises the following errors:
         - [`~job_runners.config.parquet_and_info.DatasetInBlockListError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:162]:
```suggestion
        - [`~libcommon.exceptions.DatasetInBlockListError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 3]
```diff
@@ -289,11 +226,10 @@ def raise_if_too_big_from_hub(
             The maximum size of the dataset in bytes
     Returns:
         `None`
-    <Tip>
     Raises the following errors:
         - [`~job_runners.config.parquet_and_info.DatasetTooBigFromHubError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:230]:
```suggestion
        - [`~libcommon.exceptions.DatasetTooBigFromHubError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 4]
```diff
@@ -331,13 +267,12 @@ def raise_if_too_big_from_datasets(
             The maximum size of the dataset in bytes
     Returns:
         `None`
-    <Tip>
     Raises the following errors:
-        - [`ValueError`]
-            If the datasets.config.HF_ENDPOINT is not set to the expected value
+        - [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+          If the datasets.config.HF_ENDPOINT is not set to the expected value
         - [`~job_runners.config.parquet_and_info.DatasetTooBigFromDatasetsError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:273]:
```suggestion
        - [`~libcommon.exceptions.DatasetTooBigFromDatasetsError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 5]
```diff
@@ -400,27 +335,31 @@ def raise_if_not_supported(
...
+          if the dataset does not exist, or if the token does not give the sufficient access to the dataset,
+        - [`libcommon.exceptions.GatedDisabledError`]
+          if the dataset is gated, but disabled.
+          or if the dataset is private (private datasets are not supported by the datasets server).
+        - [`libcommon.exceptions.GatedExtraFieldsError`]
+          if the dataset is gated, with extra fields. Programmatic access is not implemented for this type of
+          dataset because there is no easy way to get the list of extra fields.
+        - ['requests.exceptions.HTTPError'](https://requests.readthedocs.io/en/latest/api/#requests.HTTPError)
+          any other error when asking access
         - [`~job_runners.config.parquet_and_info.DatasetInBlockListError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:351]:
```suggestion
        - [`~libcommon.exceptions.DatasetInBlockListError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 6]
```diff
@@ -400,27 +335,31 @@ def raise_if_not_supported(
...
           If the dataset is in the list of blocked datasets.
-        - [`~libcommon.dataset.GatedExtraFieldsError`]: if the dataset is gated, with extra fields.
-            Programmatic access is not implemented for this type of dataset because there is no easy
-            way to get the list of extra fields.
-        - [`~libcommon.dataset.GatedDisabledError`]: if the dataset is gated, but disabled.
-        - [`~libcommon.dataset.DatasetNotFoundError`]: if the dataset does not exist, or if the
-            token does not give the sufficient access to the dataset, or if the dataset is private
-            (private datasets are not supported by the datasets server)
-        - ['~requests.exceptions.HTTPError']: any other error when asking access
         - [`~job_runners.config.parquet_and_info.DatasetRevisionNotFoundError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:353]:
```suggestion
        - [`~libcommon.exceptions.DatasetRevisionNotFoundError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 7]
```diff
@@ -400,27 +335,31 @@ def raise_if_not_supported(
...
           If the revision does not exist or cannot be accessed using the token.
+        - [`~job_runners.config.parquet_and_info.DatasetTooBigFromDatasetsError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:355]:
```suggestion
        - [`~libcommon.exceptions.DatasetTooBigFromDatasetsError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 8]
```diff
@@ -400,27 +335,31 @@ def raise_if_not_supported(
...
+          If the dataset is too big to be converted to parquet, as measured by the sum of the configs
+          sizes given by the datasets library.
         - [`~job_runners.config.parquet_and_info.DatasetTooBigFromHubError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:358]:
```suggestion
        - [`~libcommon.exceptions.DatasetTooBigFromHubError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py [hunk 9]
```diff
@@ -741,33 +631,35 @@ def compute_config_parquet_and_info_response(
...
+          if the dataset is gated, but disabled.
+          or if the dataset is private (private datasets are not supported by the datasets server).
+        - [`libcommon.exceptions.GatedExtraFieldsError`]
+          if the dataset is gated, with extra fields. Programmatic access is not implemented for this type of
+          dataset because there is no easy way to get the list of extra fields.
+        - ['requests.exceptions.HTTPError'](https://requests.readthedocs.io/en/latest/api/#requests.HTTPError)
+          any other error when asking access
+        - [`libcommon.simple_cache.CachedArtifactError`]
+            If the previous step gave an error.
         - [`~job_runners.config.parquet_and_info.DatasetInBlockListError`]
```
Comment (AndreaFrancis) [inline services/worker/src/worker/job_runners/config/parquet_and_info.py:649]: I think almost all the exceptions were moved to libcommon.exceptions ? There are still some comments pointing to job_runners.config.parquet_and_info.

**Review by AndreaFrancis — APPROVED** [review]
Thanks! LGTM.
I put some comments but non related to the refactor, most of them to doc still pointing to job_runner exceptions

File: services/api/src/api/utils.py (hunk 1 above)
Comment (severo) [reply services/api/src/api/utils.py:15]: yes, for coherence. Let's do it in another PR.

File: services/worker/src/worker/job_runners/config/parquet_and_info.py (hunk 2 above)
Comment (severo) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py:162]: You're right. I fix a small detail (no `~` prefix):

```suggestion
        - [`libcommon.exceptions.DatasetInBlockListError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py (hunk 3 above)
Comment (severo) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py:230]: same 

```suggestion
        - [`libcommon.exceptions.DatasetTooBigFromHubError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py (hunk 4 above)
Comment (severo) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py:273]:
```suggestion
        - [`libcommon.exceptions.DatasetTooBigFromDatasetsError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py (hunk 6 above)
Comment (severo) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py:353]:
```suggestion
        - [`libcommon.exceptions.DatasetRevisionNotFoundError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py (hunk 7 above)
Comment (severo) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py:355]:
```suggestion
        - [`libcommon.exceptions.DatasetTooBigFromDatasetsError`]
```

File: services/worker/src/worker/job_runners/config/parquet_and_info.py (hunk 9 above)
Comment (severo) [reply services/worker/src/worker/job_runners/config/parquet_and_info.py]: thanks. Fixed in https://github.com/huggingface/datasets-server/pull/1169/commits/60c9ea625af69c2c4790623dd021ffb189fe39f6

**Pull request merged**